packages/atclient/src/atclient.c

```diff
@@ -497,11 +704,39 @@ static int atclient_pkam_authenticate_validate_arguments(const atclient *ctx, co
     goto exit;
   }
 
-  if (atsign == NULL || strlen(atsign) == 0) {
+  if (NULL == atsign || strlen(atsign) == 0) {
```

Comment:
revisit this change

Reply:
change made as part of commit: [ac675cf](https://github.com/atsign-foundation/at_c/pull/423/commits/ac675cf72fe01483c980613bd39e186792bc865d)

Reply:
The point of revisiting this change was not to figure out when it was made, but why we got a segfault before it was changed. Masking the problem is not fixing the problem.